edge-hub/watchdog/src/child.rs

```diff
@@ -0,0 +1,101 @@
+use anyhow::{Context, Result};
+use nix::{
+    sys::signal::{self, Signal},
+    unistd::Pid,
+};
+use std::{
+    process::{Child, Command, Stdio},
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        Arc,
+    },
+    thread,
+    thread::{sleep, JoinHandle},
+    time::Duration,
+};
+use tracing::{error, info};
+
+const PROCESS_SHUTDOWN_TOLERANCE_SECS: Duration = Duration::from_secs(45);
```

Comment:
This is coupled to iotedged's shutdown tolerance for the edgehub container. The watchdog shutdown tolerance needs to be a bit shorter than iotedged's to make sure it can reap the child pids. 

I now see a problem where someone configures iotedged's shutdown tolerance to be super low. Iotedged will kill root docker process then the kernel will kill orphaned children leaving them as zombies. The same thing would happen if we were using a supervisor. I'm going to look for a way around this.

Reply:
So this isn't a super big deal because we can just make this time configurable. @dmolokanov Do you agree?

Reply:
Reading through this thread I am not sure if it is an issue. The poster says that the processes are removed from the process table, but a commenter seems to think it is still possible for these to become zombies.

https://stackoverflow.com/questions/39739658/what-happens-to-other-processes-when-a-docker-containers-pid1-exits

Reply:
Crisis averted:
https://stackoverflow.com/questions/49162358/docker-init-zombies-why-does-it-matter/49162867

_Next, a container is a cgroup of processes running in their own PID namespace. This cgroup is cleaned up when the container is stopped. Any zombies that are in a container are removed on stop. They don't reach the hosts init._

_Third is the different ways containers are used. Most run one main process and nothing else. If there is another process spawned it is usually a child of that main process. So until the parent exits, the zombie will exist. Then see point 2 (the zombies will be cleared on container exit_